test/SlowTests/Issues/RavenDB_14917.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Linq;
+using FastTests;
+using Orders;
+using Raven.Client.Documents.Indexes.TimeSeries;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace SlowTests.Issues
+{
+    public class RavenDB_14917 : RavenTestBase
+    {
+        public RavenDB_14917(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [Fact]
+        public void CanUseSegmentSummaryInIndex()
+        {
+            using (var store = GetDocumentStore())
+            {
+                new TimeSeriesIndex().Execute(store);
+
+                using (var session = store.OpenSession())
+                {
+                    var company = new Company();
+
+                    session.Store(company);
+
+                    session.TimeSeriesFor(company, "TS").Append(DateTime.Now, new[] { 3, 5.5 });
+                    session.TimeSeriesFor(company, "TS").Append(DateTime.Now.AddMilliseconds(10), new[] { 2, 3.5 });
+
+                    session.SaveChanges();
+                }
+
+                WaitForIndexing(store);
+
+                using (var session = store.OpenSession())
+                {
+                    var results = session
+                        .Query<TimeSeriesIndex.Result, TimeSeriesIndex>()
+                        .ToList();
+
+                    Assert.Equal(1, results.Count);
+
+                    var result = results[0];
+
+                    Assert.Equal(new[] { 2, 3.5 }, result.Min);
+                    Assert.Equal(new double[] { 5, 9 }, result.Sum);
+                    Assert.Equal(new [] { 3, 5.5 }, result.Max);
+                    Assert.Equal(2, result.Count);
+                    Assert.Equal(3, result.FirstMax);
+                    Assert.Equal(3.5, result.LastMin);
+                }
+            }
+        }
+
```

Comment:
Can we have more tests for that? With empty segments and some deleted values?
And bit more complicated query will be nice :)
Maybe something like that:
```
from segment in ts
where segment.Min[0] > 10
select new {
     segment.Entries.Where(e => e.Values[0] > 10)
}
```

Reply:
We do not iterate over empty segments and deleted values. We had other tests for checking that IIRC.